File: 442-find-all-duplicates-in-an-array/442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        //if(nums.empty())return {};
        List<Integer> ans=new ArrayList<>();
        for(int i=0;i<nums.length;i++)
        {
            if(nums[Math.abs(nums[i])-1]<0)
                ans.add(Math.abs(nums[i]));
            nums[Math.abs(nums[i])-1]=-nums[Math.abs(nums[i])-1];
        }
        return ans;
    }
}

//  List<Integer> res = new ArrayList<>();
//         for (int i = 0; i < nums.length; ++i) {
//             int index = Math.abs(nums[i])-1;
//             if (nums[index] < 0)
//                 res.add(Math.abs(index+1));
//             nums[index] = -nums[index];
//         }
//         return res;

File: 442-find-all-duplicates-in-an-array/442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> res;
        map<int,int> m;
        for(int a:nums)
            m[a]++;

      for(auto i : m )
          if(i.second==2)
              res.push_back(i.first);
        
       return res;
    }
};